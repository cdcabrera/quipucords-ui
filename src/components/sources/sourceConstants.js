import React from 'react';
import { translate } from '../i18n/i18n';
// import { dictionary } from '../../constants/dictionaryConstants';
import { ViewToolbarFieldFilter } from '../viewToolbar/viewToolbarFieldFilter';

const SourceFilterFields = [
  {
    title: () => translate('toolbar.label', { context: ['option', 'search_by_name'] }),
    value: 'search_by_name',
    component: function SearchName(props) {
      return <ViewToolbarFieldFilter queryType="search_by_name" {...props} />;
    },
    selected: true
  },
  {
    title: () => translate('toolbar.label', { context: ['option', 'search_credentials_by_name'] }),
    value: 'search_credentials_by_name',
    component: function SearchName(props) {
      return <ViewToolbarFieldFilter queryType="search_credentials_by_name" {...props} />;
    }
  }
  /*
  {
    id: 'search_by_name',
    title: 'Name',
    placeholder: 'Filter by Name',
    filterType: 'text'
  },
  {
    id: 'search_credentials_by_name',
    title: 'Credential',
    placeholder: 'Filter by Credential Name',
    filterType: 'text'
  },
  {
    id: 'source_type',
    title: 'Source Type',
    placeholder: 'Filter by Source Type',
    filterType: 'select',
    filterValues: [
      { title: dictionary.network, id: 'network' },
      { title: dictionary.satellite, id: 'satellite' },
      { title: dictionary.vcenter, id: 'vcenter' }
    ]
  }
  */
];

/**
 * ID: Enum with the following possible values [name, source_type, most_recent_connect_scan__start_time]
 */
const SourceSortFields = [
  {
    title: () => translate('toolbar.label', { context: ['option', 'name'] }),
    value: 'name',
    isNumeric: false,
    selected: true
  },
  {
    title: () => translate('toolbar.label', { context: ['option', 'source_type'] }),
    value: 'source_type',
    isNumeric: false
  },
  {
    title: () => translate('toolbar.label', { context: ['option', 'most_recent'] }),
    value: 'most_recent_connect_scan__start_time',
    isNumeric: true,
    sortAscending: false
  }
];

export { SourceFilterFields, SourceSortFields };
