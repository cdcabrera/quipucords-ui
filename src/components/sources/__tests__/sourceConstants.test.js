import { SourceFilterFields, SourceSortFields } from '../sourceConstants';

describe('SourceTypes', () => {
  it('Should have specific sourceFilterFields properties', () => {
    expect(SourceFilterFields).toMatchSnapshot('sourceFilterFields');
  });

  it('Should have specific sourceSortFields properties', () => {
    expect(SourceSortFields).toMatchSnapshot('sourceSortFields');
  });
});
