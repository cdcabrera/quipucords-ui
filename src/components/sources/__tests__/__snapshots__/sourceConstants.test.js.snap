// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SourceTypes Should have specific sourceFilterFields properties: sourceFilterFields 1`] = `
Array [
  Object {
    "filterType": "text",
    "id": "search_by_name",
    "placeholder": "Filter by Name",
    "title": "Name",
  },
  Object {
    "filterType": "text",
    "id": "search_credentials_by_name",
    "placeholder": "Filter by Credential Name",
    "title": "Credential",
  },
  Object {
    "filterType": "select",
    "filterValues": Array [
      Object {
        "id": "network",
        "title": "Network",
      },
      Object {
        "id": "satellite",
        "title": "Satellite",
      },
      Object {
        "id": "vcenter",
        "title": "VCenter",
      },
    ],
    "id": "source_type",
    "placeholder": "Filter by Source Type",
    "title": "Source Type",
  },
]
`;

exports[`SourceTypes Should have specific sourceSortFields properties: sourceSortFields 1`] = `
Array [
  Object {
    "id": "name",
    "isNumeric": false,
    "title": "Name",
  },
  Object {
    "id": "source_type",
    "isNumeric": false,
    "title": "Source Type",
  },
  Object {
    "id": "most_recent_connect_scan__start_time",
    "isNumeric": true,
    "sortAscending": false,
    "title": "Most Recent",
  },
]
`;
